src/posix/platform/hdlc_interface.cpp

```diff
@@ -656,6 +667,94 @@ void HdlcInterface::HandleHdlcFrame(otError aError)
     }
 }
 
+otError HdlcInterface::ResetSocket(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    SuccessOrExit(error = WaitForUsbDeviceRenumeration(mRadioUrl.GetPath()));
+
+    CloseFile();
+
+    mSockFd = OpenFile(mRadioUrl);
+    VerifyOrExit(mSockFd != -1, error = OT_ERROR_FAILED);
+
+exit:
+    return error;
+}
+
+otError HdlcInterface::WaitForUsbDeviceRenumeration(const char *aRadioUrlPath)
+{
+    otError error = OT_ERROR_NONE;
+
+#ifdef OPENTHREAD_USE_UDEV
+    struct udev *udev = udev_new();
+    if (!udev)
+    {
+        otLogCritPlat("Can't create udev");
+        return OT_ERROR_FAILED;
+    }
+
+    struct udev_monitor *mon = udev_monitor_new_from_netlink(udev, "udev");
+    udev_monitor_filter_add_match_subsystem_devtype(mon, "tty", NULL);
+    udev_monitor_enable_receiving(mon);
+    int fd = udev_monitor_get_fd(mon);
+
+    // wait maximally 3 seconds
+    uint64_t end = otPlatTimeGet() + 3000 * US_PER_MS;
+    uint64_t now;
+
+    do
+    {
+        fd_set         fds;
+        struct timeval tv;
+
+        tv.tv_sec  = 0;
+        tv.tv_usec = 100 * US_PER_MS;
+
+        FD_ZERO(&fds);
+        FD_SET(fd, &fds);
+
+        int ret = select(fd + 1, &fds, NULL, NULL, &tv);
+        if (ret > 0 && FD_ISSET(fd, &fds))
+        {
+            struct udev_device *dev = udev_monitor_receive_device(mon);
+            if (dev)
+            {
+                struct udev_list_entry *entry;
+                const char *            action = udev_device_get_action(dev);
+
+                if (strncmp(action, "add", strlen("add")) == 0)
+                {
+                    udev_list_entry_foreach(entry, udev_device_get_devlinks_list_entry(dev))
+                    {
+                        const char *name = udev_list_entry_get_name(entry);
+                        if (strncmp(name, aRadioUrlPath, strlen(aRadioUrlPath)) == 0)
+                        {
+                            udev_device_unref(dev);
+                            goto exit;
+                        }
+                    }
+                }
+                udev_device_unref(dev);
+            }
+        }
+        now = otPlatTimeGet();
+    } while (end > now);
+
+    error = OT_ERROR_FAILED;
+
+exit:
+    udev_unref(udev);
+
+#else
+    (void)aRadioUrlPath;
```

Comment:
Use `OT_UNUSED_VARIABLE()` instead.

The `(void)aVar` does not work on all tool-chains so we have `OT_UNUSED_VARIABLE()` for this purpose.